gateway/utils/metrics/helpers.go

```diff
@@ -2,106 +2,97 @@ package metrics
+func (m *Module) createFileDocuments(key string, metrics *metricOperations, t string) []interface{} {
 	docs := make([]interface{}, 0)
 
-	dbMetrics.Range(func(key, value interface{}) bool {
-		parts := strings.Split(key.(string), ":")
-		metrics := value.(*crudMetrics)
+	arr := strings.Split(key, ":")
+	module := "file"
+	if metrics.create > 0 {
+		docs = append(docs, m.createDocument(arr[0], arr[1], "na", module, utils.Create, metrics.create, t))
+	}
 
-		if metrics.create > 0 {
-			docs = append(docs, m.createCrudDocument(project, parts[0], parts[1], utils.Create, metrics.create, t))
-		}
+	if metrics.read > 0 {
+		docs = append(docs, m.createDocument(arr[0], arr[1], "na", module, utils.Read, metrics.read, t))
+	}
 
-		if metrics.read > 0 {
-			docs = append(docs, m.createCrudDocument(project, parts[0], parts[1], utils.Read, metrics.read, t))
-		}
+	if metrics.delete > 0 {
+		docs = append(docs, m.createDocument(arr[0], arr[1], "na", module, utils.Delete, metrics.delete, t))
+	}
 
-		if metrics.update > 0 {
-			docs = append(docs, m.createCrudDocument(project, parts[0], parts[1], utils.Update, metrics.update, t))
-		}
+	if metrics.list > 0 {
+		docs = append(docs, m.createDocument(arr[0], arr[1], "na", module, utils.List, metrics.list, t))
+	}
 
-		if metrics.delete > 0 {
-			docs = append(docs, m.createCrudDocument(project, parts[0], parts[1], utils.Delete, metrics.delete, t))
-		}
+	return docs
+}
 
-		if metrics.batch > 0 {
-			docs = append(docs, m.createCrudDocument(project, parts[0], parts[1], utils.Batch, metrics.batch, t))
-		}
+func (m *Module) createCrudDocuments(key string, value *metricOperations, t string) []interface{} {
+	docs := make([]interface{}, 0)
 
-		// Delete the key from the map
-		dbMetrics.Delete(key)
+	arr := strings.Split(key, ":")
+	module := "db"
+	if value.create > 0 {
+		docs = append(docs, m.createDocument(arr[0], arr[1], arr[2], module, utils.Create, value.create, t))
+	}
 
-		return true
-	})
+	if value.read > 0 {
+		docs = append(docs, m.createDocument(arr[0], arr[1], arr[2], module, utils.Read, value.read, t))
+	}
 
-	return docs
-}
+	if value.update > 0 {
+		docs = append(docs, m.createDocument(arr[0], arr[1], arr[2], module, utils.Update, value.update, t))
+	}
 
-func (m *Module) createCrudDocument(project, dbType, col string, op utils.OperationType, count uint64, t *time.Time) interface{} {
-	return map[string]interface{}{
-		"id":         ksuid.New().String(),
-		"project_id": project,
-		"module":     "db",
-		"type":       op,
-		"sub_type":   col,
-		"ts":         *t,
-		"count":      count,
-		"driver":     dbType,
-		"node_id":    "sc-" + m.nodeID,
+	if value.delete > 0 {
+		docs = append(docs, m.createDocument(arr[0], arr[1], arr[2], module, utils.Delete, value.delete, t))
 	}
+
+	if value.batch > 0 {
+		docs = append(docs, m.createDocument(arr[0], arr[1], arr[2], module, utils.Batch, value.batch, t))
```

Comment:
We shouldn't have a separate batch type. Instead let count the number of update, create and delete in each batch